### PR TITLE
fix(autoid): add autoid to ContentGroup CTA

### LINKED ISSUE #1884: Content Group Simple is missing an data-autoid attribute for optional CTA
### The problem
_{{The Content Group Simple has an optional CTA.  We do not have a way to exclude all of the CTA markup.  There is a data-autoid on dds--card but that leaves unused markup.   We are currently using the one on dds--card  now so this is not a MVP blocker but needs to be cleaned up in the future )}}_

### The solution
_{{Add data-autoid to where the class bx-content-group__cta-row  is defined so we can exclude content based on it when no CTA is included.}}_

### Additional information
 _{{Please provide any additional information or resources for reference}}_
 - _{{user research}}_
 - _{{user insights}}_
 - _{{user metrics}}_

### PR DESCRIPTION
**Pull request by kennylam**
### Related Ticket(s)

#1747 #1884

### Description

Adds `data-autoid` to `ContentGroup`'s optional CTA

### Changelog

**New**

- `data-autoid` to `ContentGroup` CTA

### CODE REVIEW

File: packages/react/src/patterns/sub-patterns/ContentGroup/ContentGroup.js [hunk 1]
```diff
@@ -44,7 +44,9 @@ const ContentGroup = ({ children, heading, customClassName, cta }) => {
         {children}
       </div>
       {cta && (
-        <div className={`${prefix}--content-group__cta-row`}>
+        <div
+          data-autoid={`${stablePrefix}--content-group_cta}`}
```
Comment (jeffchew) [inline packages/react/src/patterns/sub-patterns/ContentGroup/ContentGroup.js:48]: @kennylam can you also add to the README?

Comment (kennylam) [reply packages/react/src/patterns/sub-patterns/ContentGroup/ContentGroup.js:48]: @jeffchew Right! Forgot about that.

File: packages/react/src/patterns/sub-patterns/ContentGroup/README.md [hunk 2]
```diff
@@ -70,6 +70,7 @@ Add the following line on your `.env` file at the root of your project,
 | Name                           | Description                   |
 | ------------------------------ | ----------------------------- |
 | `dds--content-group`           | Content group wrapper element |
+| `dds--content-group_cta`       | Content group CTA element     |
```
Comment (jeffchew) [inline packages/react/src/patterns/sub-patterns/ContentGroup/README.md:73]: @kennylam it looks weird that this is one underscore instead of two...

Comment (kennylam) [reply packages/react/src/patterns/sub-patterns/ContentGroup/README.md:73]: oops... typo let me fix that.

File: packages/react/src/patterns/sub-patterns/ContentGroup/ContentGroup.js (hunk 1 above)
Comment (jeffchew) [inline packages/react/src/patterns/sub-patterns/ContentGroup/ContentGroup.js:48]: @kennylam should change here too right?

Comment (kennylam) [reply packages/react/src/patterns/sub-patterns/ContentGroup/ContentGroup.js]: @jeffchew Holy moly what is going on with me today?

**Pull request merged**